tests/api/clever/test_clever.py

```diff
@@ -67,20 +74,30 @@ def test_external_type_from_clever_grade(self):
             assert external_type_from_clever_grade(none_grade) is None
 
 
-class TestCleverAuthenticationAPI(DatabaseTest):
-    def setup_method(self):
-        super().setup_method()
-        self.api = MockAPI(self._default_library, self.mock_integration)
+class CleverAuthenticationFixture:
+    db: DatabaseTransactionFixture
+    api: MockAPI
+    app: Flask
+
+    @classmethod
+    def create(cls, db: DatabaseTransactionFixture) -> CleverAuthenticationFixture:
+        fixture = CleverAuthenticationFixture()
+        fixture.db = db
+        fixture.api = MockAPI(db.default_library(), fixture.mock_integration())
+
         os.environ["AUTOINITIALIZE"] = "False"
         from api.app import app
 
+        fixture.app = app
+
+        return fixture
+
+    def close(self):
         del os.environ["AUTOINITIALIZE"]
```

Comment:
To be pedantic, it should probably restore the old value of the environment variable here. We don't actually know if it was unset before we started, so we should probably leave it how we found it.

Reply:
Er, what I mean is:

1. Save the old value
2. Set the new value to `False`
3. Restore the old value in `close()`

Reply:
The `monkeypatch` fixture that pytest supplies, automates the process for patching env variables. Maybe we want to use that here instead of doing this ourselves? 

https://docs.pytest.org/en/6.2.x/monkeypatch.html#monkeypatching-environment-variables

something like: 

```python
def clever_fixture(
    db: DatabaseTransactionFixture, monkeypatch: MonkeyPatch
):
    fixture = CleverAuthenticationFixture.create(db)
    monkeypatch.setenv("AUTOINITIALIZE", "False")
    yield fixture
    fixture.close()
```

Reply:
Monkeypatching is neat 👍  I went with that in https://github.com/ThePalaceProject/circulation/pull/825/commits/b8bb780e9a1ee0129c6668f7e01154814f0d8e1c and removed the create/yield/close/del pattern and reverted back to something similar to the first version, just using monkeypatch with it.